File: flaskdemo1/app/dataFun/node/companyNode.py
from ..data1 import *

def search(category,name):
    qstr = "match (n:company) where 1=1 "
    if category != '':
        qstr += 'and n.category="' + category + '" '
    if name != '':
        qstr += 'and n.name="' + name + '" '

    qstr = qstr + ' return n'

    print(qstr)
    return query_graph(qstr)

def add(category,name,location,mainBusiness):
    qstr = 'merge (n:company{'\
        'name:"' + name + '", ' \
        'category:"'+category+'",' \
        'location:"'+location+'",' \
        'mainBusiness:"'+mainBusiness+'"' \
        '})  return n'
    print(qstr)
    return query_graph(qstr)

def delete(id):
    print(id)
    qstr = "match (n:company) where 1=1 "
    if id != '':
        qstr += 'and id(n)='+id+' '
    else:
        return False
    qstr += ' detach delete n'

    print(qstr)
    return query_graph(qstr)


def update(id,name):
    qstr = 'match (m)-[r:雇佣]-(n:company) where id(n)='+id+' return r'
    if len(query_graph(qstr)['nodes'])==0:
        qstr = 'match (n:company) where id(n)='+id+' '
        setq = 'set n.name="' + name + '" return n'
    else:
        qstr = 'match (m)-[r:雇佣]-(n:address) where id(n)=' + id + ' '
        setq = 'set n.name="' + name + '",m.home="' + name + '"  return n'

    # str = 'match (m)-[r:居住]-(n:address) where id(n)='+id+' '
    # set = 'set n.name="' + name + '",m.home="'+name+'"  return n'
    # if fkind != kind:
    #     str = 'match (n:' + fkind + ') where id(n)='+id+' '
    #     set = 'remove n:'+fkind+' set n:'+kind+''
    # else:
    #     set += 'n.name="'+name+'" ,n.age='+age+' ,n.gender="'+gender+'" ,n.degree="'+degree+'" ,n.work="'+work+'",n.work="'+work+'"  return n'

    qstr += setq
    print(qstr)
    return query_graph(qstr)

File: flaskdemo1/static/ajax/node/companyNode.js
function search() {
    var category = document.getElementById("category").value;
    var name = document.getElementById("name").value;
    var location = document.getElementById("detailAddress").value;
    var mainBusiness = document.getElementById("mainBusiness").value;
    $.ajax({
        url: "/node/csearch",
        type:"POST",
        dataType: "json",
        data:{
            category:category,
            name:name,
            location:location,
            mainBusiness:mainBusiness,
        },
        success:function (data) {
            //console.log(data);
            //console.log('调用了1');
            if (data['nodes']==0){
                layer.msg('没找到该数据!', {icon: 5});
            }
            else {
                mychart.setOption(option,true);
                //console.log(data);
                mychart.setOption({
                    series:[{
                        data:data['nodes'],
                        links:data['links'],
                    },
                    ]
                });
                }


        }
    })
}

function add() {
    var category = document.getElementById("category").value;
    var name = document.getElementById("name").value;
    var location = document.getElementById("detailAddress").value;
    var mainBusiness = document.getElementById("mainBusiness").value;

    if (name==""||category==''||location==''||mainBusiness==''){
        layer.msg('所有信息不能为空!', {icon: 5});
        return false;
    }

    $.ajax({
        url:"/node/cadd",
        type:"POST",
        data:{
            category:category,
            name:name,
            location:location,
            mainBusiness:mainBusiness,
        },
        success:function (data) {
            console.log(data);
            if (data=="hasNode"){
                layer.msg('已存在该节点!', {icon: 5});
                return false;
            }
            else if (data=="noLocation"){
                layer.msg('地址不存在，请检查!', {icon: 5});
                return false;
            }
            else if (data=="locationHasTaken"){
                layer.msg('地址已被其他公司注册，请检查!', {icon: 5});
                return false;
            }
            else if(data=="success"){
                layer.msg('创建成功!', {icon: 1});
                search();
                return;
            }
            else{
                layer.msg('创建失败!', {icon: 1});
                return;
            }
        }
    })
}

function del() {

    var id = document.getElementById('nid').title;
    var category = document.getElementById("category").value;
    var name = document.getElementById("name").value;
    var location = document.getElementById("detailAddress").value;
    var mainBusiness = document.getElementById("mainBusiness").value;

    $.ajax({
        url:"/node/cdelete",
        type:"POST",
        data:{
            id:id,
            category:category,
            name:name,
            location:location,
            mainBusiness:mainBusiness,
        },
        success:function (data) {
            console.log(data);
            if (data=="fail"){
                layer.msg('删除失败!', {icon: 5});
                //location.replace(document.referrer);
            }
            else {
                layer.msg('删除成功!', {icon: 1});
                //location.replace(document.referrer);
            }
        }
    })
}

function update() {
    var id = document.getElementById('nid').title;
    // var fkind = document.getElementById('fkind').title;
    // var kind = document.getElementById("kind").value;
    var name = document.getElementById("name").value;
    // var age = document.getElementById("age").value;
    // var gender = document.getElementById("gender").value;
    // var degree = document.getElementById("degree").value;
    // var work = document.getElementById("work").value;
    // var home = document.getElementById("home").value;||age==""||gender==""||degree==""||work==""||home==""

    if (id==''){
        layer.msg('请点击图中节点!', {icon: 5});
        return false;
    }

    if (name==""){
        layer.msg('请完善要修改的信息!', {icon: 5});
        return false;
    }

    $.ajax({
        url:'/node/cupdate',
        type:'POST',
        data:{
            id:id,
            // fkind:fkind,
            // kind:kind,
            name:name,
            // age:age,
            // gender:gender,
            // degree:degree,
            // work:work,
            // home:home,
        },
        success:function (data) {
            console.log(data);
            if (data=="fail"){
                layer.msg('修改失败!', {icon: 5});
            }
            else {
                layer.msg('修改成功!', {icon: 1});
                search();
            }
        }
    })

}
